File: redsheet-portal/src/app/database-models/v-knowledge-gathering-plan-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vKnowledgeGatheringPlanRecord {
    constructor(vKnowledgeGatheringPlanRecord?: vKnowledgeGatheringPlanRecord )  {
        let properties = [ "knowledgeGatheringPlanId", "knowledgeGivenBy", "knowledgeGivenOn", "knowledgeRequired", "projectPowerId", "source",];
        for (let property of properties)
            if (vKnowledgeGatheringPlanRecord && vKnowledgeGatheringPlanRecord[property])
                this[property] = vKnowledgeGatheringPlanRecord[property];
    }
 
	knowledgeGatheringPlanId : number =   0 ;
 
    @required()
    @maxLength(1000)
	knowledgeGivenBy : string =   undefined;
 
    @required()
	knowledgeGivenOn : Date =   undefined;
 
    @required()
    @maxLength(1000)
	knowledgeRequired : string =   undefined;
 
    @range(1,2147483647)
	projectPowerId : number =   undefined;
 
    @required()
    @maxLength(150)
	source : string =   undefined;


}

File: RedSheet.DbEntities/Models/vKnowledgeGatheringPlanRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vKnowledgeGatheringPlanRecords", Schema = "dbo")]
	[Serializable]
	public partial class vKnowledgeGatheringPlanRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int KnowledgeGatheringPlanId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string KnowledgeGivenBy { get; set; } 
 
	[Required]
	public System.DateTime KnowledgeGivenOn { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string KnowledgeRequired { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectPowerId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Source { get; set; } 

	public vKnowledgeGatheringPlanRecord()
	{
    }

    }
}

